lib/algos/list/tools.js

```diff
@@ -31,11 +32,49 @@ function inc(str) {
             String.fromCharCode(str.charCodeAt(str.length - 1) + 1)) : str;
 }
 
+/*
+ * Compares two Strings, s1 and s2, using UTF-8 lexicographic ordering.
+ * @function
+ * @param {String} s1 the first string to compare
+ * @param {String} s2 the second string to compare
+ * @return {number} -1, 0, or 1 if s1 is less than, equal or greater than s2 respectively
+ */
+function utf8Compare(s1, s2) {
+    const l1 = s1.length;
+    const l2 = s2.length;
+    const l = Math.min(l1, l2)
+
+    for (let i = 0; i < l; i++) {
+        let cp1 = s1.codePointAt(i);
```

Comment:
You could declare these as `const`s

Reply:
Sure, I'm open to suggestions here. Not sure what the best practice is TBH. I haven't touched Javascript for 5-6 years and back then var was all you had.

Reply:
const is for anything that doesn't get reassigned ( pushing to a const array is fine ). Let is for values that do change, such as `i` here. Var is basically unused at this point and considered bad practice. Eslint might fail you for it depending on settings I think.